src/AppBundle/Repository/PostRepository.php

```diff
@@ -36,8 +43,17 @@ public function queryLatest()
         ;
     }
 
-    public function findLatest()
+    /**
+     * @param int $page
+     *
+     * @return Pagerfanta
+     */
+    public function findLatest($page = 1)
     {
-        $this->queryLatest()->getResult();
+        $paginator = new Pagerfanta(new DoctrineORMAdapter($this->queryLatest(), true, false));
```

Comment:
technically, there is no fetch joined collection in this query, so you can optimize the pagination by changing the second argument